bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/cli/CliProvider.java

```diff
@@ -220,12 +220,12 @@ public Optional<ConfigValue> getAsUnionAndMark(Module module, VariableKey key) {
         CliArg cliArg = getCliArg(module, key);
         BUnionType unionType = (BUnionType) ((BIntersectionType) key.type).getEffectiveType();
         boolean isEnum = SymbolFlags.isFlagOn(unionType.getFlags(), SymbolFlags.ENUM);
-        if (!isEnum && ConfigUtils.containsUnsupportedMembers(unionType)) {
-            throw new ConfigException(CONFIG_CLI_TYPE_NOT_SUPPORTED, key.variable, unionType);
-        }
         if (cliArg.value == null) {
             return Optional.empty();
         }
+        if (!isEnum && ConfigUtils.containsUnsupportedMembers(unionType)) {
```

Comment:
Shall we add some tests and modify existing tests accordingly?

Reply:
I'll do that.